File: 프로그래머스/Lv2/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

//각 크기마다 개수 저장할 배열 -> 정렬 후 개수 많은 순으로 살펴보기

bool cmp(int a, int b){
    return a>b;
}

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    int arr[10000001];
    fill(arr,arr+10000001,0);
    
    for(int i=0;i<tangerine.size();i++){
        arr[tangerine[i]]++;
    }
    
    sort(arr,arr+10000001,cmp);
    int Index=0;
    
    while(k>0){
        k-=arr[Index++];
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/Lv2/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    d={}
    
    for i in tangerine:
        if i in d:
            d[i]=d[i]+1
        else:
            d[i]=1
    
    newD=dict(sorted(d.items(), key=lambda x: x[1], reverse=True))
    
    for i in newD:
        k-=newD[i]
        answer+=1
        if(k<=0): break
    
    return answer
